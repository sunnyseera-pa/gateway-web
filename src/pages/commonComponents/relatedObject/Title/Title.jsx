/** @jsx jsx */
import { jsx } from '@emotion/react';
import React from 'react';
import PropTypes from 'prop-types';
import '../../CommonComponents.scss';
import * as styles from '../Dataset/Dataset.styles';

const Title = props => {
    const { type, name, id, className, onClickHandler, activeLink } = props;
    return activeLink ? (
        <>
            <a
                onClick={onClickHandler}
                className={className ? className : 'purple-bold-16'}
                css={styles.pointer}
                href={`/${type}/${id}`}
                data-testid={`title-${type}-${id}`}
            >
                {name}
            </a>
            {props.children}
        </>
    ) : (
        <>
            <span className={className ? className : 'black-bold-16'} data-testid={`title-${type}-${id}`}>
                {' '}
                {name}
            </span>
            {props.children}
        </>
    );
};

Title.propTypes = {
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    activeLink: PropTypes.bool,
    onClickHandler: PropTypes.func,
    className: PropTypes.string,
};

Title.defaultProps = {
    activeLink: false,
    type: 'title',
};

export default Title;
