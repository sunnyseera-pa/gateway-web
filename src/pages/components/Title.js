
// /ShowObjects/Title.js
import React, { Component } from 'react';
import axios from 'axios';

class Title extends Component {

  constructor(props) {
    super(props)
    this.state.id = props.id;
    this.state.type = props.type;
    this.state.name = props.name;
    this.state.description = props.description;
    this.state.rating = props.rating;
    this.state.link = props.link;
  }

  // initialize our state
  state = {
    id: '',
    type: '',
    name: '',
    description: '',
    rating: '',
    link: ''
  };


  // here is our UI
  // it is easy to understand their functions when you
  // see them render into our screen
  render() {
    const { id, type, name, description, rating, link } = this.state;
    return (
      <div className="Rectangle">
        <p>Title = {id} = {type}</p>

        
      </div>
    );
  }
}

export default Title;