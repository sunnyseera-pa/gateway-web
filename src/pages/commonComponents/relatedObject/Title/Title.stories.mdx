import { Meta, Story, Canvas } from '@storybook/addon-docs';
import Title from './Title';

<Meta
	title='Related Resources'
	component={Title}
	args={{
		name: 'this is a test title',
		type: 'dataset',
		id: '123',
	}}
/>

<Canvas>
	<Story name='Title'>{props => <Title {...props} />}</Story>
</Canvas>
