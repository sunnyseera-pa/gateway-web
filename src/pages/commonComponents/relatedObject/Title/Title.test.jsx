import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Title from './Title';

const props = {
    name: 'This is a test description',
    id: '123',
    type: 'test',
};
let wrapper;

describe('Given the Title component', () => {
    describe('When it is rendered', () => {
        beforeAll(() => {
            wrapper = render(<Title {...props} />);
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        it('Then description should be rendered', () => {
            expect(screen.getByTestId(`title-${props.type}-${props.id}`)).toHaveTextContent(props.name);
        });
    });
    describe('When activeLink is true', () => {
        it('Then Title should be clickable', () => {
            const { rerender } = wrapper;
            rerender(<Title {...props} activeLink={true} />);
            expect(screen.getByTestId(`title-${props.type}-${props.id}`)).toHaveAttribute('href', `/${props.type}/${props.id}`);
        });
    });
});
