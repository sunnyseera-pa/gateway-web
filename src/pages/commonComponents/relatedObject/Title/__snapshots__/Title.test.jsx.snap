// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Title component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <span
    class="black-bold-16"
    data-testid="title-test-123"
  >
     
    This is a test description
  </span>
</div>
`;
